### PR TITLE
MES-4297 Test Report Flow

### PR DESCRIPTION
**Pull request by tomcrawleyy**
## Description
NOTE: This has not been tested thoroughly due to the page currently being inaccessible from inside of the application.
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/test-report/cat-d/test-report.cat-d.page.ts [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+} from 'ionic-angular';
+import { Store, select } from '@ngrx/store';
+import { Observable } from 'rxjs/Observable';
+import { merge } from 'rxjs/observable/merge';
+import { map } from 'rxjs/operators';
+import { Subscription } from 'rxjs/Subscription';
+import { AuthenticationProvider } from '../../../providers/authentication/authentication';
+import { StoreModel } from '../../../shared/models/store.model';
+import { getUntitledCandidateName } from '../../../modules/tests/journal-data/common/candidate/candidate.selector';
+import { getCandidate } from '../../../modules/tests/journal-data/cat-c/candidate/candidate.cat-c.reducer';
```
Comment (matthew2564) [inline src/pages/test-report/cat-d/test-report.cat-d.page.ts:19]: leave TODO for candidate.cat-d.reducer

File: src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    it('should call the end test function', () => {
+      spyOn(component, 'onEndTestClick');
+      const endTestButton = fixture.debugElement.query(By.css('#end-test-button'));
+      endTestButton.triggerEventHandler('click', null);
+      expect(component.onEndTestClick).toHaveBeenCalled();
+    });
+  });
+
+  describe('showUncoupleRecouple', () => {
+    it('should return false when the test is a category C', () => {
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts]: should be D

File: src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      component.testCategory = TestCategory.D;
+      expect(component.showUncoupleRecouple()).toBeFalsy();
+    });
+
+    it('should return false when the test is a category C1', () => {
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts]: should be D1

File: src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      component.testCategory = TestCategory.D1;
+      expect(component.showUncoupleRecouple()).toBeFalsy();
+    });
+
+    it('should return true when the test is a category C1E', () => {
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts]: should be D1E

File: src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      component.testCategory = TestCategory.D1E;
+      expect(component.showUncoupleRecouple()).toBeTruthy();
+    });
+
+    it('should return true when the test is a category CE', () => {
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts]: Should be DE

File: src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+import { NavigationStateProviderMock } from '../../../../providers/navigation-state/__mocks__/navigation-state.mock';
+import { candidateMock } from '../../../../modules/tests/__mocks__/tests.mock';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { configureTestSuite } from 'ng-bullet';
+// TODO Implement Cat D UncoupleRecoupleComponent
+import { UncoupleRecoupleComponent } from '../../cat-c/components/uncouple-recouple/uncouple-recouple';
+// TODO Implement Cat D ReverseLeftComponent
+import { ReverseLeftComponent } from '../../cat-c/components/reverse-left/reverse-left';
+// TODO Implement Cat D ReverseLeftPopoverComponent
+import { ReverseLeftPopoverComponent } from '../../cat-c/components/reverse-left-popover/reverse-left-popover';
```
Comment (TomBillingtonUK) [inline src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts:48]: These should be copied over as part of the pr the category specific components are part of the page.

Comment (tomcrawleyy) [reply src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts:48]: Uncouple/Recouple and VehicleChecksComponent haven't been created yet, although the Cat D for Reversing gets copied over in the next PR

Comment (TomBillingtonUK) [reply src/pages/test-report/cat-d/__tests__/test-report.cat-d.page.spec.ts:48]: they should be included in this PR, The copy is of the page and it's custom components.

**Pull request closed without merging**